media/js/mozorg/contribute/taskview.js

```diff
@@ -0,0 +1,273 @@
+$(function() {
+    'use strict';
+
+    var $taskSteps = $('.task-steps');
+    var $stepOne = $('#step_one');
+    var $thankYou = $('#thankyou');
+    var $getFirefox = $('.get-firefox');
+    var $actionButton = $('.action');
+    var $downloadButton = $('.download-link');
+
+    var $followButton = $('.follow-mozilla')
+    var intentURL = $followButton.attr('href');
+
+    var $signupTweetForm = $('#signup-tweet');
+
+    // some tasks, like installing Whimsy, required the user to be using Firefox
+    if ($getFirefox.length > -1 && !Mozilla.Client.isFirefox) {
+        $getFirefox.toggleClass('hidden');
+        // hide any action buttons that forms part of the task.
+        $actionButton.addClass('hidden');
+    }
+
+    /**
+     * Toggles the completed class on the relevant step, and calls
+     * taskComplete() if this was the final step of the task.
+     */
+    function completeStep($step) {
+        var $stepOne = $('#step_one');
+        var $stepTwo = $('#step_two');
+
+        if ($step.data('step') === 'one') {
+            $stepOne.toggleClass('completed');
+        } else if ($step.data('step') === 'two') {
+            $stepTwo.toggleClass('completed');
+        }
+
+        if ($step.data('complete') === true) {
+            taskComplete();
+        }
+    }
+
+    /**
+     * Called once all steps of the task has been completed. This will
+     * show the thank message and scroll it into view.
+     */
+    function taskComplete() {
+        $thankYou.removeClass('visibly-hidden');
+        $thankYou.attr('aria-hidden', 'false');
+        $thankYou[0].scrollIntoView();
+        $thankYou.focus();
+    }
+
+    /**
+     * Waits for the initial tab/window to become visible, and then
+     * proceeds to complete the relevant task step.
+     */
+    function handleVisibilityChange($step) {
+        $(document).on('visibilitychange.taskview', function() {
```

Comment:
For browsers that don't support the `visibilitychange` event, this listener will never get unbound again since it won't fire. To avoid a build up of duplicate event listeners would it be better to call `off` first before calling `on`?